packages/app/pages/Api/organization.tsx

```diff
@@ -0,0 +1,113 @@
+/**
+ * This exports a React element which displays a list of all organizations,
+ * and provides a section to select a single organization out of that list
+ *
+ * @package: REA app
+ * @author:  Steven Fontaine <fontainesw@msoe.edu>
+ * @since:   2018-02-08
+ */
+
+import * as React from "react";
+import "./api.css";
+import getAllOrganizations from "../../../ui-bindings/Organization/getAllOrganizations";
+import { concatArray } from "./agent";
+import getOrganizationById from "../../../ui-bindings/Organization/getOrganizationById";
+
+export const Organization = (props) => {
+  let organization = props.organization;
+  return(
+    <div>
+      <div>id: {organization.id}</div>
+      <div>name: {organization.name}</div>
+      <div>type: {organization.type}</div>
+      <div>image: {organization.image}</div>
+      <div>note: {organization.note}</div>
+      <div>ownedEconomicResources: {concatArray(organization.ownedEconomicResources)}</div>
+      <div>agentProcesses: {concatArray(organization.agentProcesses)}</div>
+      <div>agentPlans: {concatArray(organization.agentPlans)}</div>
+      <div>agentEconomicEvents: {concatArray(organization.agentEconomicEvents)}</div>
+      <div>agentCommitments: {concatArray(organization.agentCommitments)}</div>
+      <div>agentRelationships: {concatArray(organization.agentRelationships)}</div>
+      <div>agentRoles: {concatArray(organization.agentRoles)}</div>
+      <div>agentRecipess: {concatArray(organization.agentRecipes)}</div>
+      <br/>
+    </div>
+  );
+};
+
+const OrganizationField = (props) => {
+  return(
+    <div>
+      <form onSubmit={props.onSubmitAction}>
+        Enter an Id: <input type="text" name="value" onChange={props.setOrganization}/>
+        <input type="submit" value="query"/>
+      </form>
+    </div>
+  );
+};
+
+export const GetAllOrganizations = getAllOrganizations(({ organizationList, loading, error}) => {
+  return (
+    loading ? <strong>Loading...</strong> : (
+      error ? <p style={{color: "#F00"}}>API error</p> : (
+        <div>
+          {concatArray(organizationList)}
+        </div>
+      )
+    )
+  );
+});
+
+export const GetSingleOrganization = getOrganizationById(({ organization, loading, error }) => {
+  return (
+    loading ? <strong>Loading...</strong> : (
```

Comment:
Same as above, I would remove the ternary operator.

Reply:
resolved